RallyHereDebugTool/Source/Private/RH_ImGuiUtilities.cpp

```diff
@@ -166,129 +166,123 @@ void ImGuiDisplayCustomData(const TMap<FString, FString>& CustomData, const FStr
-			}
-			else
-			{
-				ImGui::Text("%s", TCHAR_TO_UTF8(*ValueStr));
-			}
+		FGuid TestGuid;
+
+		if (FGuid::Parse(ValueStr, TestGuid))
+		{
+			ImGui::Text("%s", TCHAR_TO_UTF8(*TestGuid.ToString(EGuidFormats::DigitsWithHyphens)));
 		}
-		else if (FMapProperty const* MapProp = CastField<FMapProperty>(Property))
+		else
+		{
+			ImGui::Text("%s", TCHAR_TO_UTF8(*ValueStr));
+		}
+	}
+	else if (FMapProperty const* MapProp = CastField<FMapProperty>(Property))
+	{
+		if (ImGui::TreeNodeEx(TCHAR_TO_UTF8(*FString::Printf(TEXT("Data##%s"), *Key)), RH_DefaultTreeFlags))
 		{
-			if (ImGui::TreeNodeEx(TCHAR_TO_UTF8(*FString::Printf(TEXT("Data##%s"), *Key)), RH_DefaultTreeFlags))
+			if (ImGui::BeginTable("DataTable", 2, RH_TableFlagsPropSizing))
 			{
-				if (ImGui::BeginTable("DataTable", 2, RH_TableFlagsPropSizing))
-				{
-					ImGui::TableSetupColumn("Key");
-					ImGui::TableSetupColumn("Value");
-					ImGui::TableHeadersRow();
+				ImGui::TableSetupColumn("Key");
+				ImGui::TableSetupColumn("Value");
+				ImGui::TableHeadersRow();
 
-					FScriptMapHelper MapHelper(MapProp, Property->ContainerPtrToValuePtr<uint8>(&Model));
-					for (int32 i = 0; i < MapHelper.Num(); ++i)
-					{
-						FString	KeyStr;
-						FString	ValueStr;
-						MapHelper.GetKeyProperty()->ExportText_Direct(KeyStr, MapHelper.GetKeyPtr(i), MapHelper.GetKeyPtr(i), nullptr, PPF_None);
-						MapHelper.GetValueProperty()->ExportText_Direct(ValueStr, MapHelper.GetValuePtr(i), MapHelper.GetValuePtr(i), nullptr, PPF_None);
+				FScriptMapHelper MapHelper(MapProp, Data);
+				for (int32 i = 0; i < MapHelper.Num(); ++i)
+				{
+					FProperty* KeyProp = MapHelper.GetKeyProperty();
+					FProperty* ValueProp = MapHelper.GetValueProperty();
 
-						ImGui::TableNextRow();
-						ImGui::TableNextColumn();
-						ImGui::Text(TCHAR_TO_UTF8(*KeyStr));
-						ImGui::TableNextColumn();
-						ImGui::Text(TCHAR_TO_UTF8(*ValueStr));
-					}
+					ImGui::TableNextRow();
+					ImGui::TableNextColumn();
+					ImGuiDisplayProperty(KeyProp->GetName(), KeyProp, nullptr, MapHelper.GetPairPtr(i));
+					ImGui::TableNextColumn();
+					ImGuiDisplayProperty(ValueProp->GetName(), ValueProp, nullptr, MapHelper.GetPairPtr(i));
+				}
 
-					ImGui::EndTable();
+				ImGui::EndTable();
 
-				}
-				ImGui::TreePop();
 			}
+			ImGui::TreePop();
 		}
-		else if (FStructProperty const* StructProp = CastField<FStructProperty>(Property))
+	}
+	else if (FStructProperty const* StructProp = CastField<FStructProperty>(Property))
+	{
+		if (ImGui::TreeNodeEx(TCHAR_TO_UTF8(*FString::Printf(TEXT("Data##%s"), *Key)), RH_DefaultTreeFlags))
 		{
-			if (ImGui::TreeNodeEx(TCHAR_TO_UTF8(*FString::Printf(TEXT("Data##%s"), *Key)), RH_DefaultTreeFlags))
-			{
-				ImGuiDisplayModelData(*(FRHAPI_Model*)StructProp->ContainerPtrToValuePtr<void>(&Model), StructProp->Struct);
-				ImGui::TreePop();
-			}
+			ImGuiDisplayModelData(*(FRHAPI_Model*)Data, StructProp->Struct);
```

Comment:
Can we make this cast safer?  Type checking in some way?  You should be able to do something like Struct->IsChildOf(FRHAPI_Model::StaticStruct()), or something.